pkg/healthcheck/healthcheck_test.go

```diff
@@ -2519,6 +2522,26 @@ func TestLinkerdIdentityCheck(t *testing.T) {
 			schemeInConfig:   string(corev1.SecretTypeTLS),
 			expectedOutput:   []string{"linkerd-identity-test-cat certificate config is valid: key ca.crt containing the trust anchors needs to exist in secret linkerd-identity-issuer if --identity-external-issuer=true"},
 		},
+		{
+			checkerToTest:    "certificate config is valid",
+			checkDescription: "does not get influenced by newline differences between trust anchors",
+			tlsSecretScheme:  string(corev1.SecretTypeTLS),
+			schemeInConfig:   string(corev1.SecretTypeTLS),
+			expectedOutput:   []string{"linkerd-identity-test-cat certificate config is valid"},
+			configAnchorsModifier: func(anchors string) string {
+				return strings.TrimSpace(anchors)
+			},
+		},
+		{
+			checkerToTest:    "certificate config is valid",
+			checkDescription: "fails when roots in config and secret do not match",
+			tlsSecretScheme:  string(corev1.SecretTypeTLS),
+			schemeInConfig:   string(corev1.SecretTypeTLS),
+			expectedOutput:   []string{"linkerd-identity-test-cat certificate config is valid: IdentityContext.TrustAnchorsPem does not match ca.crt in linkerd-identity-issuer"},
+			configAnchorsModifier: func(anchors string) string {
+				return "nonsense"
```

Comment:
In order to actually test that extra newlines at the end of the certs don't affect things, you might wanna change this with something like `return anchors + "\n"` and make it a positive test...

Reply:
:+1: